README.md

```diff
@@ -12,10 +12,63 @@ The `SHR` opcode is 3 gas cheaper than `DIV` and also bypasses Solidity's divisi
 
 ```solidity
 // Unoptimized:
-uint256 two = 4 / 2;
+uint256 foo = bar / 2;
 
 // Optimized:
-uint256 two = 4 >> 1;
+uint256 foo = bar >> 1;
 ```
 
+
+### Use `>=` and `<=` instead of `>` and `<`
+
+Explanation tbd
+
+- [Gas Usage]()
+- [Full Example]()
+
+```solidity
+// Unoptimized:
+bool foo = bar > 0;
+
+// Optimized:
+bool foo = bar >= 1;
+```
+
+
+### Using `!=` is usually cheaper than `>` or `<`
+
+Explanation tbd
+
+- [Gas Usage]()
+- [Full Example]()
+
+```solidity
+// Unoptimized:
+uint256 foo = bar < 32;
+
+// Optimized:
+uint256 foo = bar != 32;
+```
+
+
+### Use `<address>.code.length` instead of assembly `extcodesize` to avoid variable overhead
+
+Solidity [0.8.1](https://github.com/ethereum/solidity/blob/develop/Changelog.md#081-2021-01-27) implements a shortcut for `<address>.code.length` that avoids copying code to memory.  More explanation TBD
```

Comment:
well using extcodesize in assembly avoids copying to mem already, .code.length is just cheaper if u use inline cuz then u avoid extra stack ops